src/mono_item.rs

```diff
@@ -16,6 +16,18 @@ impl<'gcc, 'tcx> PreDefineMethods<'tcx> for CodegenCx<'gcc, 'tcx> {
         let ty = instance.ty(self.tcx, ty::ParamEnv::reveal_all());
         let gcc_type = self.layout_of(ty).gcc_type(self, true);
 
+        let is_mutable = self.tcx.is_mutable_static(def_id);
+
+        // As an optimization, all shared statics which do not have interior
+        // mutability are placed into read-only memory.
+        let gcc_type =
+            if !is_mutable && self.type_is_freeze(ty) {
+                gcc_type.make_const()
+            }
+            else {
+                gcc_type
+            };
```

Comment:
The libgccjit12 CI failed and I have the feeling it could be due to this change.
Can you please try to feature-gate this as well?

Reply:
Done in 42dee9393c7f945c6805140f879f56a6a07b6149